vm/src/interpreter.cpp

```diff
@@ -87,17 +97,19 @@ void run(constant_pool pool, int ip, const char *bytes, size_t size) {
             // Fork and exec the process
             pid_t pid = fork();
             if (pid == 0) {
-                // Execute the process
+                // Replace the current process with a new process image
                 execvp(argv[0], argv);
             } else {
                 for (int i = 0; i < frame_size; i++) {
                     delete[] argv[i];
                 }
                 delete[] argv;
+                int status = 0;
                 // Wait for the process to finish
-                waitpid(pid, nullptr, 0);
-                // Pop the stack
-                stack_frame -= frame_size;
+                waitpid(pid, &status, 0);
+
+                // add status to the stack
+                stack.push_int(status);
```

Comment:
```suggestion
                // Add the exit status to the stack
                stack.push_int(WEXITSTATUS(status));
```
Inserting the status will always result in a truthy value.